auth/api/iam/api_mitzxnuts.go

```diff
@@ -0,0 +1,464 @@
+
+			body, err := io.ReadAll(resp.Body)
+			if err != nil {
+				return nil, err
+			}
+
+			err = json.Unmarshal(body, &definition)
+			if err != nil {
+				return nil, err
+			}
+		} else {
+			return nil, errors.New("presentation_definition or presentation_definition_uri is missing")
+		}
+
+	}
+	return &definition, nil
+}
+
+func (r Wrapper) CallbackOid4vciIssuance(ctx context.Context, request CallbackOid4vciIssuanceRequestObject) (CallbackOid4vciIssuanceResponseObject, error) {
+
+	sessionId := request.SessionId
+	code := request.Params.Code
+
+	oid4vciSession := Oid4vciSession{}
+	err := r.getSessionStore("oid4vci").Get(sessionId, &oid4vciSession)
+	if err != nil {
+		return nil, err
+	}
+
+	pkceParams := PKCEParams{}
+	err = r.getSessionStore("pkce").Get(sessionId, &pkceParams)
+	if err != nil {
+		return nil, err
+	}
+
+	issuerDid, err := did.ParseDID(oid4vciSession.IssuerDid)
+	holderDid, err := did.ParseDID(oid4vciSession.HolderDid)
+	metadata, err := r.auth.MitzXNutsIAMClient().OpenIdCredentialIssuerMetadata(ctx, *issuerDid)
+	if len(metadata.AuthorizationServers) == 0 {
+		return nil, core.NotFoundError("cannot locate any authorization endpoint in %s", issuerDid.String())
+	}
+	for i := range metadata.AuthorizationServers {
+		serverURL, err := url.Parse(metadata.AuthorizationServers[i])
+		if err != nil {
+			return nil, err
+		}
+		metadataFromUrl, err := r.auth.MitzXNutsIAMClient().OpenIdConfiguration(ctx, *serverURL)
+		if err != nil {
+			return nil, err
+		}
+
+		tokenEndpoint := metadataFromUrl.TokenEndpoint
+		response, err := r.auth.MitzXNutsIAMClient().AccessTokenOid4vci(ctx, holderDid.String(), tokenEndpoint, oid4vciSession.RedirectUri, code, &pkceParams.Verifier)
+		println(response.AccessToken)
+
+		proofJwt, err := r.proofJwt(ctx, *holderDid, *issuerDid, nil)
+		if err != nil {
+			return nil, err
+		}
+
+		credentials, err := r.auth.MitzXNutsIAMClient().VerifiableCredentials(ctx, metadata.CredentialEndpoint, response.AccessToken, proofJwt)
+		if err != nil {
+			return nil, err
+		}
+		credential, err := vc.ParseVerifiableCredential(credentials.Credential)
+		if err != nil {
+			return nil, err
+		}
+		for t := range credential.Type {
+			trusted, err := r.vcr.Trusted(credential.Type[t])
+			if err != nil {
+				return nil, err
+			}
+			if !slices.Contains(trusted, credential.Issuer) {
+				err := r.vcr.Trust(credential.Type[t], credential.Issuer)
+				if err != nil {
+					return nil, err
+				}
+			}
+
+		}
+
+		err = r.vcr.Verifier().Verify(*credential, true, true, nil)
+		if err != nil {
+			return nil, err
+		}
+		err = r.vcr.Wallet().Put(ctx, *credential)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	return CallbackOid4vciIssuance302Response{
+		Headers: CallbackOid4vciIssuance302ResponseHeaders{Location: oid4vciSession.RedirectUrl},
+	}, nil
+}
+
+func (r Wrapper) proofJwt(ctx context.Context, issuerDid did.DID, audienceDid did.DID, additionalClaims *map[string]interface{}) (string, error) {
```

Comment:
could be moved to client or even wallet to make use of signing capabilities.

Reply:
Confirmed, moved to client.